### ISSUE TITLE
There is a bug when you put 0.0 for val_ratio in RadnomLinkSplit

### ISSUE THREAD

**Issue by Nemanja3214** [pyg-team/pytorch_geometric#10590]
### 🐛 Describe the bug

```python
    transform = T.Compose([T.NormalizeFeatures(),
                           # T.ToUndirected(),
                           T.RandomLinkSplit(
                                num_val=0.0,
                                num_test=0.2,
                                is_undirected=True,
                                add_negative_train_samples=False,
                                disjoint_train_ratio=0.0,
                                neg_sampling_ratio=5,
                                edge_types=[pt],
                                # rev_edge_types=[rev_pt]
                            )
                           ])
````
edge_index for both train_data and test_data stays the same and there is data leakage between edges in training thereforth

### Versions

aiohappyeyeballs          2.6.1
aiohttp                   3.13.2
aiosignal                 1.4.0
attrs                     25.4.0
beautifulsoup4            4.14.3
bleach                    6.3.0
certifi                   2025.11.12
charset-normalizer        3.4.4
colorama                  0.4.6
contourpy                 1.3.3
cycler                    0.12.1
defusedxml                0.7.1
dnspython                 2.8.0
fastjsonschema            2.21.2
filelock                  3.20.0
fonttools                 4.61.1
frozenlist                1.8.0
fsspec                    2025.12.0
hf-xet                    1.2.0
huggingface-hub           0.36.0
idna                      3.11
intel-openmp              2021.4.0
Jinja2                    3.1.6
joblib                    1.5.2
jsonschema                4.25.1
jsonschema-specifications 2025.9.1
jupyter_client            8.7.0
jupyter_core              5.9.1
jupyterlab_pygments       0.3.0
kiwisolver                1.4.9
MarkupSafe                3.0.3
matplotlib                3.10.8
mistune                   3.1.4
mkl                       2021.4.0
mpmath                    1.3.0
multidict                 6.7.0
nbclient                  0.10.2
nbconvert                 7.16.6
nbformat                  5.10.4
networkx                  3.6.1
numpy                     1.26.4
packaging                 25.0
pandocfilters             1.5.1
pillow                    12.1.0
pip                       25.1.1
platformdirs              4.5.1
propcache                 0.4.1
psutil                    7.1.3
Pygments                  2.19.2
pymongo                   4.15.5
pyparsing                 3.2.5
python-dateutil           2.9.0.post0
PyYAML                    6.0.3
pyzmq                     27.1.0
rdflib                    7.5.0
referencing               0.37.0
regex                     2025.11.3
requests                  2.32.5
rpds-py                   0.30.0
safetensors               0.7.0
scikit-learn              1.8.0
scipy                     1.16.3
sentence-transformers     5.2.0
setuptools                80.9.0
six                       1.17.0
soupsieve                 2.8
sympy                     1.13.1
tbb                       2021.13.1
threadpoolctl             3.6.0
tinycss2                  1.4.0
tokenizers                0.22.1
torch                     2.2.2+cu121
torch_cluster             1.6.3+pt22cu121
torch_geometric           2.5.3
torch_scatter             2.1.2+pt22cu121
torch_sparse              0.6.18+pt22cu121
torch_spline_conv         1.2.2+pt22cu121
torchaudio                2.2.2+cu121
torchvision               0.17.2+cu121
tornado                   6.5.3
tqdm                      4.67.1
traitlets                 5.14.3
transformers              4.57.3
typing_extensions         4.15.0
urllib3                   2.6.2
webencodings              0.5.1
xxhash                    3.6.0
yarl                      1.22.0
